trio/_core/tests/test_windows.py

```diff
@@ -38,5 +41,109 @@
             print("end loop")
 
 
+async def test_WaitForMultipleObjects_sync():
+    # One handle
+    handle1 = kernel32.CreateEventA(ffi.NULL, True, False, ffi.NULL)
+    t = Thread(target=WaitForMultipleObjects_sync, args=(handle1,))
+    t.start()
+    kernel32.SetEvent(handle1)
+    t.join()  # the test succeeds if we do not block here :)
+    kernel32.CloseHandle(handle1)
+
+    # Two handles, signal first
+    handle1 = kernel32.CreateEventA(ffi.NULL, True, False, ffi.NULL)
+    handle2 = kernel32.CreateEventA(ffi.NULL, True, False, ffi.NULL)
+    t = Thread(target=WaitForMultipleObjects_sync, args=(handle1, handle2))
+    t.start()
+    kernel32.SetEvent(handle1)
+    t.join()  # the test succeeds if we do not block here :)
+    kernel32.CloseHandle(handle1)
+    kernel32.CloseHandle(handle2)
+
+    # Two handles, signal seconds
+    handle1 = kernel32.CreateEventA(ffi.NULL, True, False, ffi.NULL)
+    handle2 = kernel32.CreateEventA(ffi.NULL, True, False, ffi.NULL)
+    t = Thread(target=WaitForMultipleObjects_sync, args=(handle1, handle2))
+    t.start()
+    kernel32.SetEvent(handle2)
+    t.join()  # the test succeeds if we do not block here :)
+    kernel32.CloseHandle(handle1)
+    kernel32.CloseHandle(handle2)
+
+    # Closing the handle will not stop the thread. Initiating a wait on a
+    # closed handle will fail/return, but closing a handle that is already
+    # being waited on will not stop whatever is waiting for it.
+
+
+async def test_WaitForSingleObject():
+
+    # Set the timeout used in the tests. The resolution of WaitForSingleObject
+    # is 0.01 so anything more than a magnitude larger should probably do.
+    # If too large, the test become slow and we might need to mark it as @slow.
+    TIMEOUT = 0.5
```

Comment:
The comment about the resolution is out of date, right?

If this is a test that has to sleep – and I think it probably is – then we're probably going to mark it `@slow` :-). I try to keep the quick test run you do over and over while hacking as fast as possible, so right now the slowest single test is <0.3s. We always run the full test suite during CI runs though, so all the `@slow` tests still get run on every commit.

Maybe we could have one test that's not marked `@slow` and checks the trivial cases of (1) `WaitForSingleObject` on an already-ready object, (2) `WaitForSingleObject` that gets preemptively cancelled, and then a second test that is marked `@slow` and has the tests for (3) we wait a bit and then signal the object, (4) we wait a bit and then cancel the call?

Reply:
Yes, that comment was outdated.